### PR TITLE
Bump AGP to v7.0.0

### PR DESCRIPTION
**Pull request by msasikanth**
- Bump AGP to v7.0.0
- Bump Gradle wrapper to v7.0.2
- Replace deprecated Gradle usages
- Disable Lint partial analysis
- Enable checking dependencies when running lint
- Enable core library desugaring in platform module
- Remove unused resources in router module
- Bump Lint to v30.0.0
- Add `Vendor` in `SimpleIssueRegistry`
- Update CHANGELOG

**Story**: https://app.clubhouse.io/simpledotorg/story/4326/bump-agp-to-v7-0-0

**Release notes**: https://developer.android.com/studio/releases/gradle-plugin#7-0-0

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -193,6 +193,7 @@ android {
     isWarningsAsErrors = true
     isAbortOnError = true
     isCheckReleaseBuilds = false
+    isCheckDependencies = true
```
Comment (vinaysshenoy) [inline app/build.gradle.kts:196]: What does this setting do?

File: simple-platform/build.gradle.kts [hunk 2]
```diff
@@ -22,9 +22,15 @@ android {
       )
     }
   }
+
+  compileOptions {
```
Comment (vinaysshenoy) [inline simple-platform/build.gradle.kts:26]: Hmm, why do we need to turn this on specifically for this module? Doesn't turning it on in the main `app` module automatically take care of it for submodules as well?

**Review by vinaysshenoy — APPROVED** [review]

File: app/build.gradle.kts (hunk 1 above)
Comment (msasikanth) [reply app/build.gradle.kts:196]: This ensures that lint is also ran on our modules.

File: simple-platform/build.gradle.kts (hunk 2 above)
Comment (msasikanth) [reply simple-platform/build.gradle.kts:26]: Turns out, it doesn't. Not sure why it was working before.
https://developer.android.com/studio/write/java8-support

image

**Pull request merged**